manifests/profile/networking/firewall.pp

```diff
@@ -21,8 +24,74 @@
     ensure => absent
   }
 
-  resources { 'firewall':
-    purge => true,
+  if $internal_routing == '' {
+    resources { 'firewall':
+      purge => true,
+    }
+  } else {
+    case $internal_routing {
+      'docker': {
+        $input_ignore = [
+        ]
+
+        $output_ignore = [
+          '-j DOCKER',
+        ]
+
+        $forward_ignore = [
+          '-j DOCKER-USER',
+          '-j DOCKER-ISOLATION',
+          '-i docker0',
+          '-o docker0',
+        ]
+      }
+
+      'kubernetes_calico': {
+        $input_ignore = [
+          '-j cali-INPUT',
+          '-j KUBE-FIREWALL',
+          '-j KUBE-SERVICES',
+          '-j KUBE-EXTERNAL-SERVICES',
+        ]
+
+        $output_ignore = [
+          '-j cali-OUTPUT',
+          '-j KUBE-FIREWALL',
+          '-j KUBE-SERVICES',
+        ]
+
+        $forward_ignore = [
+          '-j cali-FORWARD',
+          '-j KUBE-FORWARD',
+          '-j KUBE-SERVICES',
+        ]
+      }
+
+      default: {
+        $input_ignore = []
+        $output_ignore = []
+        $forward_ignore = []
```

Comment:
In this default case, we'll only manage the INPUT, OUTPUT, and FILTER chains, and we'll purge everything on them that we didn't put there. Seems like a pretty safe default to me?

Reply:
Agreed.